tools/parse_soundfont.py

```diff
@@ -1107,14 +1115,22 @@ def process_percussion_set(self):
                 oot_drum.envelope = drum.envelope_enum
                 oot_drum.pan = max(0, min(64 + round(1.27 * drum.pan), 127))
                 oot_drum.index = index
-                oot_drum.release_index = ctypes.c_int8(find_closest_index(drum.release, release_values)).value
+                oot_drum.release_index = ctypes.c_uint8(find_closest_index(drum.release, release_values)).value # release rates are UNSIGNED integers, not SIGNED integers owl...
                 oot_drum.name = f"drum_{index}"
                 oot_drum.enum = f"DRUM_{index}"
                 #tuning logic
-                pseudorootkey = drum.rootkey - 9 - index + 60
+                # Key-based instruments always assume a root is 60, so just subtract the key range from the root
+                # to get the distance from 60 to act as a secondary coarse tune value...
+                # Unless there's some weird shenanigans that require the "Zelda" note values, this should result
+                # in the proper tuning float value being calculated...
+                #
+                # There is no reason to subtract "X" or to add 60, the key-based tuning calc doesn't inverse the
+                # resultant like the channel-based tuning calculation does... Adding 60 here doesn't cancel out
+                # the 60 in channel-based tuning either, because the 60 is being ADDED to the negative result...
+                pseudorootkey = drum.rootkey - index
                 #print(f"root key: {pseudorootkey}")
-                oot_drum.tuningfloat = calculate_inst_tuning(pseudorootkey, 
-                                       drum.tuning_semi,
+                oot_drum.tuningfloat = calc_keybased_tuning(pseudorootkey,
```

Comment:
this generates the drums array decrementing in pitch not incrementing, the games code doesn't multiply the tuning float here against the float of the key being played but rather just resamples it based on just the float here. so the tuning should increment in value